src/dotnet/APIView/APIViewWeb/Repositories/ReviewManager.cs

```diff
@@ -427,5 +439,28 @@ private async Task AssertAutomaticReviewModifier(ClaimsPrincipal user, ReviewMod
                 throw new AuthorizationFailedException();
             }
         }
+
+        private async Task<ReviewRevisionModel> FindMatchingApprovedRevision(ReviewModel review)
+        {
+            var revisionModel = review.Revisions.LastOrDefault();
+            var revisionFile = revisionModel.Files.FirstOrDefault();
+            var codeFile = await _codeFileRepository.GetCodeFileAsync(revisionModel);
+
+            // Get manual reviews to check if a matching review is in approved state
+            var reviews = await _reviewsRepository.GetReviewsAsync(revisionFile.Language, revisionFile.PackageName, false);
+            foreach (var r in reviews)
+            {
+                var approvedRevision = r.Revisions.Where(r => r.Approvers.Count() > 0).LastOrDefault();
```

Comment:
Why `LastOrDefault` and not all the revisions?

Reply:
Reason is to compare against latest revision of approved review rather than against all approved revisions which will be a lot of processing when comparing against old one unnecessarily. In most cases, automatically created master review should match with latest approved revision.